src/actions/actions.ts

```diff
@@ -269,12 +284,42 @@ export abstract class BaseCommand extends BaseAction {
    * Run the command the number of times VimState wants us to.
    */
   public async execCount(position: Position, vimState: VimState): Promise<VimState> {
-    let timesToRepeat = this.canBePrefixedWithCount ? vimState.recordedState.count || 1 : 1;
+    if (!this.runsOnceForEveryCursor()) {
+      let timesToRepeat = this.canBePrefixedWithCount ? vimState.recordedState.count || 1 : 1;
+
+      for (let i = 0; i < timesToRepeat; i++) {
+        vimState = await this.exec(position, vimState);
+      }
+
+      return vimState;
+    }
 
-    for (let i = 0; i < timesToRepeat; i++) {
-      vimState = await this.exec(position, vimState);
+    let timesToRepeat  = this.canBePrefixedWithCount ? vimState.recordedState.count || 1 : 1;
+    let resultingCursors: Range[] = [];
+    let i              = 0;
+
+    const cursorsToIterateOver = vimState.allCursors
```

Comment:
I actually explained what was going on over here: 

https://github.com/VSCodeVim/Vim/pull/811/files#diff-a7723f32fe6b05e135838f84e5c23c0fR965

Unfortunately I had to have similar logic in 2 places but I think the alternative is even worse. :P